File: Solution/Python/357. Count Numbers with Unique Digits.py
class Solution:
    def countNumbersWithUniqueDigits(self, n):
        """
        :type n: int
        :rtype: int
        """
        used = [False for x in range(10)]

        def gen(used, n, d):
            if n == d:
                return 1
            total = 1
            startIndex = 1 if d == 0 else 0
            for i in range(startIndex, 10):
                if not used[i]:
                    used[i] = True
                    total += gen(used, n, d + 1)
                    used[i] = False
            return total

        return gen(used, n, 0)


s = Solution()
a = s.countNumbersWithUniqueDigits(7)
print(a)

File: Solution/Javascript/357. Count Numbers with Unique Digits.js
/**
 * @param {number} n
 * @return {number}
 */
var countNumbersWithUniqueDigits = function (n) {
    let res = 0;
    let gen = (s, n) => {
        if (n == 0) {
            res += 1;
        } else {
            for (let i in "0123456789") {
                if (i == '0' || !s.includes(i)) {
                    gen(s + i, n - 1)
                }
            }
        }
    }
    gen("", n)
    return res;
};

let res = countNumbersWithUniqueDigits(2)
console.log(res)